### PR TITLE
Updates from review comments from PR #21076

### PR DESCRIPTION
**Pull request by jhanders34**
- Update the Facade creation to only be on the impls that need it instead of in the Abstract that doesn't have a need to reference it.
- Remove IHttpSession and its implementation logic from the different classes since it isn't needed.  The security info is never set since it is tWAS specific logic.
- Remove getSessionData method and just use the field it returns since it is already protected and available to the subclasses.
- Remove HttpSessionFacadeImpl since it isn't used anywhere.
- Remove SessionData60 since it has no value now.
- Add `@Override`'s and finals in some of the classes to make it cleaner.
- Other small clean up to remove compile warnings about deprecation and missing generic references.

#build

### CODE REVIEW

**Comment by pmd1nh** [comment]
Full Session, WC FATs

#build      #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

File: dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java [hunk 1]
```diff
@@ -36,14 +36,10 @@
  */
 public abstract class AbstractSessionData extends AbstractHttpSession implements IBMSessionExt {
 
-    protected static final String SECURITY_PROP_NAME = "com_ibm_ejs_security_httpsession_info";
-
-    protected boolean _hasSecurityInfo = true; // used in
     // getAttributeNames/getValueNames
```
Comment (pmd1nh) [inline dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java]: delete these comments as well.

Comment (jhanders34) [reply dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractSessionData.java]: done

**Comment by pmd1nh** [comment]
Full Session FATs had one failure in com.ibm.ws.sesson.cache_fat bucket. I resubmitted for that single build and all passed

com.ibm.ws.session.cache_fat FAT tests | 182 | 0 | 1 |   |  
-- | -- | -- | -- | -- | --
com.ibm.ws.session.cache_fat FAT tests | 182 | 0 | 0 |

**Review by pmd1nh — APPROVED** [review]

**Comment by pmd1nh** [comment]
#libby

**Review by pnicolucci — APPROVED** [review]

**Comment by pnicolucci** [comment]
@zhang-ca just a heads up if you want to see the changes being made to the session component.

**Review by zhang-ca — APPROVED** [review]
Reviewed the changes. Thanks for the notification.

**Pull request merged**